porch/controllers/rootsyncsets/pkg/controllers/rootsyncset/controller.go

```diff
@@ -108,59 +116,215 @@ func (r *RootSyncSetReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		// Stop reconciliation as the item is being deleted
 		return ctrl.Result{}, nil
 	}
-	var patchErrs []error
+
+	results := make(reconcileResult)
 	for _, clusterRef := range rootsyncset.Spec.ClusterRefs {
 		clusterRefName := clusterRef.Kind + ":" + clusterRef.Name
+		result := clusterRefResult{}
 		client, err := r.GetClient(ctx, clusterRef, rootsyncset.Namespace)
 		if err != nil {
-			patchErrs = append(patchErrs, err)
-			continue
-		}
-		rootSyncRes, newRootSync, err := BuildObjectsToApply(&rootsyncset)
-		if err != nil {
-			patchErrs = append(patchErrs, err)
+			result.clientError = err
 			continue
```

Comment:
I think we skip `results[clusterRefName] = result` in this case?  Maybe pre-assign (though we'd have to make it a pointer)?

Reply:
Good catch. Fixed.